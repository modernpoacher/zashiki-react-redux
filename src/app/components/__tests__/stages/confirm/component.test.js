import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import {
  RESOLVED,
  REJECTED,
  PENDING
} from '@modernpoacher/zashiki-react-redux/app/common'

import Confirm, { getErrorProps, getConfirmProps } from '@modernpoacher/zashiki-react-redux/app/components/stages/confirm/component'

Enzyme.configure({ adapter: new Adapter() })

jest.mock('@modernpoacher/zashiki-react-redux/app/common', () => ({
  RESOLVED: 'MOCK RESOLVED',
  REJECTED: 'MOCK REJECTED',
  PENDING: 'MOCK PENDING'
}))

jest.mock('@modernpoacher/zashiki-react-redux/app/components/stages/confirm/status/resolved')
jest.mock('@modernpoacher/zashiki-react-redux/app/components/stages/confirm/status/rejected')
jest.mock('@modernpoacher/zashiki-react-redux/app/components/stages/confirm/status/pending')

describe('@modernpoacher/zashiki-react-redux/app/components/stages/confirm/component', () => {
  const MOCK_DEFINITION = {
    meta: {},
    elements: {}
  }
  const MOCK_RESPONSE = {}
  const MOCK_ONSUBMIT = jest.fn()
  const MOCK_ONEMBARK = jest.fn()

  const mockProps = {
    definition: {},
    response: {},
    onSubmit: jest.fn(),
    onConfirm: jest.fn(),
    exception: {
      name: 'MOCK NAME',
      message: 'MOCK MESSAGE'
    }
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('Always', () => {
    const component = (
      <Confirm
        status={RESOLVED}
        definition={MOCK_DEFINITION}
        response={MOCK_RESPONSE}
        onSubmit={MOCK_ONSUBMIT}
        onConfirm={MOCK_ONEMBARK}
      />
    )

    const spy = jest.spyOn(Confirm.prototype, 'componentDidMount')

    let wrapper

    beforeEach(() => {
      wrapper = shallow(component)
    })

    it('renders', () => {
      expect(wrapper)
        .toMatchSnapshot()
    })

    it('invokes `componentDidMount`', () => {
      expect(spy)
        .toHaveBeenCalled()
    })

    it('invokes `onConfirm`', () => {
      expect(MOCK_ONEMBARK)
        .toHaveBeenCalled()
    })
  })

  describe('`getConfirmProps`', () => {
    it('is defined', () => {
      expect(getConfirmProps)
        .toBeDefined()
    })
  })

  describe('`getErrorProps`', () => {
    it('is defined', () => {
      expect(getErrorProps)
        .toBeDefined()
    })
  })

  describe('`RESOLVED`', () => {
    it('renders', () => {
      const component = (
        <Confirm
          status={RESOLVED}
          definition={MOCK_DEFINITION}
          response={MOCK_RESPONSE}
          onSubmit={MOCK_ONSUBMIT}
          onConfirm={MOCK_ONEMBARK}
        />
      )

      expect(toJson(shallow(component)))
        .toMatchSnapshot()
    })
  })

  describe('`REJECTED`', () => {
    it('renders', () => {
      const component = (
        <Confirm
          status={REJECTED}
          exception={{
            name: 'MOCK NAME',
            message: 'MOCK MESSAGE'
          }}
          response={MOCK_RESPONSE}
          onSubmit={MOCK_ONSUBMIT}
          onConfirm={MOCK_ONEMBARK}
        />
      )

      expect(toJson(shallow(component)))
        .toMatchSnapshot()
    })
  })

  describe('`PENDING`', () => {
    it('renders', () => {
      const component = (
        <Confirm
          status={PENDING}
          definition={MOCK_DEFINITION}
          response={MOCK_RESPONSE}
          onSubmit={MOCK_ONSUBMIT}
          onConfirm={MOCK_ONEMBARK}
        />
      )

      expect(toJson(shallow(component)))
        .toMatchSnapshot()
    })
  })

  describe('`getErrorProps()`', () => {
    it('returns an object', () => {
      expect(getErrorProps(mockProps))
        .toEqual({
          name: 'MOCK NAME',
          message: 'MOCK MESSAGE'
        })
    })
  })

  describe('`getConfirmProps()`', () => {
    it('returns an object', () => {
      expect(getConfirmProps(mockProps))
        .toEqual({
          definition: {},
          response: {},
          onSubmit: expect.any(Function)
        })
    })
  })
})
