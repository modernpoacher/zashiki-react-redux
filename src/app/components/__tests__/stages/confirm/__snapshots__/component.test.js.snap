// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/confirm/component \`PENDING\` renders 1`] = `<$default />`;

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/confirm/component \`REJECTED\` renders 1`] = `
<Rejected
  message="MOCK MESSAGE"
  name="MOCK NAME"
/>
`;

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/confirm/component \`RESOLVED\` renders 1`] = `
<Resolved
  definition={
    Object {
      "elements": Object {},
      "meta": Object {},
    }
  }
  onSubmit={[MockFunction]}
  response={Object {}}
/>
`;

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/confirm/component Always renders 1`] = `ShallowWrapper {}`;
